tests/adapter/dbt/tests/adapter/constraints/test_constraints.py

```diff
@@ -228,7 +228,8 @@ def test__constraints_ddl(self, project, expected_sql):
         )
 
         results = run_dbt(["run", "-s", "+my_model"])
-        assert len(results) == 2
+        # assert at least my_model was run - additional upstreams may or may not be provided to the test setup via models fixture
+        assert len(results) >= 1
```

Comment:
Why do we not know how many are provided?

Reply:
We know in BaseConstraintsRuntimeDdlEnforcement (which includes a test for foreign key, requiring an extra upstream model) that there will be 2 run models. However in subclasses that override the `models` fixture - it's not guaranteed that they will provide a foreign key model to test the foreign key constraint. 

The rest of the test assumes that `my_model` (the one we're actually testing against with constraints defined) exists, but doesn't strictly require an upstream foreign key model for setup.